File: Post-First-Year/pascals-triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        result = [[1] for _ in range(numRows)]
        for i in range(1, numRows):
            for x in range(1, i):
                result[i].append(result[i-1][x] + result[i-1][x-1])
            result[i].append(1)
        return result
        

File: Post-First-Year/pascals-triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>>pascal(numRows);
        pascal[0]= {1};
        for(int i = 1; i < numRows; i++) {
            pascal[i] = {1};
            for(int j = 1; j < i; j++) {
                pascal[i].push_back(pascal[i-1][j] + pascal[i-1][j-1]);
            }
            pascal[i].push_back(1);
        }
        return pascal;
    }
};